src/modules/rewards/services/arb-rebate-service.ts

```diff
@@ -36,12 +41,110 @@ export class ArbRebateService {
   constructor(
     @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
     @InjectRepository(ArbReward) readonly arbRewardRepository: Repository<ArbReward>,
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
     private marketPriceService: MarketPriceService,
     private ethProvidersService: EthProvidersService,
     private appConfig: AppConfig,
     private dataSource: DataSource,
   ) {}
 
+  public async getEarnedRewards(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    const data = await this.cacheManager.get(getArbEarnedRewardsCacheKey(userAddress));
+    if (data) return data;
```

Comment:
Caching the data here would not bring many improvements because the earned referral rewards are not cached, so on every request you have to wait for them to be computed. What do you think if we cache the entire response of the `/rewards/earned` endpoint, not the sub-components

Reply:
Definitely, yes! I've implemented it and also moved the cache logict to the rewards service. Lmk what u think